File: app14_LUA_UNITY/backend/db/migrations/001_create_frameworks.lua
local schema = require("lapis.db.schema")
local types = schema.types

return {
  up = function()
    schema.create_table("frameworks", {
      { "code", types.varchar({ primary_key = true }) },
      { "name", types.varchar({ null = false }) },
      { "version", types.varchar({ null = false, default = "" }) },
      { "description", types.text({ null = false, default = "" }) },
      { "reference_url", types.varchar({ null = false, default = "" }) }
    })
  end,
  down = function()
    schema.drop_table("frameworks")
  end
}

File: app13_ruby_FastApi/backend/db/migrations/001_create_frameworks.rb
# frozen_string_literal: true

Sequel.migration do
  change do
    create_table(:frameworks) do
      String :code, primary_key: true
      String :name, null: false
      String :version, null: false
      String :description, text: true, null: false, default: ""
      String :reference_url, null: false, default: ""
    end
  end
end
